sdk-workflows/src/main/java/io/dapr/workflows/client/WorkflowStatusQuery.java

```diff
@@ -0,0 +1,212 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.workflows.client;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class used for constructing orchestration metadata queries.
+ */
+public class WorkflowStatusQuery {
+
+  private List<WorkflowRuntimeStatus> runtimeStatusList = new ArrayList<>();
+  private Instant createdTimeFrom;
+  private Instant createdTimeTo;
+  private List<String> taskHubNames = new ArrayList<>();
+  private int maxInstanceCount = 100;
+  private String continuationToken;
+  private String instanceIdPrefix;
+  private boolean fetchInputsAndOutputs;
+
+  /**
+   * Sole constructor.
+   */
+  public WorkflowStatusQuery() {
+
+  }
+
+  /**
+   * Sets the list of runtime status values to use as a filter. Only orchestration instances that have a matching
+   * runtime status will be returned. The default {@code null} value will disable runtime status filtering.
+   *
+   * @param runtimeStatusList the list of runtime status values to use as a filter
+   * @return this query object
+   */
+  public WorkflowStatusQuery setRuntimeStatusList(@Nullable List<WorkflowRuntimeStatus> runtimeStatusList) {
+    this.runtimeStatusList = runtimeStatusList;
+    return this;
+  }
+
+  /**
+   * Include orchestration instances that were created after the specified instant.
+   *
+   * @param createdTimeFrom the minimum orchestration creation time to use as a filter or {@code null} to disable this
+   *                        filter
+   * @return this query object
+   */
+  public WorkflowStatusQuery setCreatedTimeFrom(@Nullable Instant createdTimeFrom) {
+    this.createdTimeFrom = createdTimeFrom;
+    return this;
+  }
+
+  /**
+   * Include orchestration instances that were created before the specified instant.
+   *
+   * @param createdTimeTo the maximum orchestration creation time to use as a filter or {@code null} 
+   *        to disable this filter
+   * @return this query object
+   */
+  public WorkflowStatusQuery setCreatedTimeTo(@Nullable Instant createdTimeTo) {
+    this.createdTimeTo = createdTimeTo;
+    return this;
+  }
+
+  /**
+   * Sets the maximum number of records that can be returned by the query. The default value is 100.
+   * 
+   * <p>Requests may return fewer records than the specified page size, even if there are more records.
+   * Always check the continuation token to determine whether there are more records.
+   *
+   * @param maxInstanceCount the maximum number of orchestration metadata records to return
+   * @return this query object
+   */
+  public WorkflowStatusQuery setMaxInstanceCount(int maxInstanceCount) {
```

Comment:
can `maxInstanceCount` be `null` as the other query parameters?
If so, consider using `Integer` as int is not nullable

Reply:
by looking at durableTask it indeed does not take nullables. I believe you can ignore this. :)